### DISCUSSION TITLE
Some files compress better with --all-normal

### DISCUSSION THREAD

**Discussion by dreamlayers** [tjko/jpegoptim#105] [General]
I just ran jpegoptim compiled with mozjpeg 3.1 on all my photos. 82 out of 15552 (around 0.5%) of files compressed better as baseline using --all-normal.

(I first ran `jpegoptim -p --strip-com --strip-xmp --strip-iptc --strip-icc` on all photos putting optimized copies in a new directory tree. Then I added --all-normal to the command line, and ran that on all optimized photos created by the first run and all original photos for which no optimized photos were created by the first run, putting files in a third directory tree. Then I counted the number of files in that third tree.)

**Comment by dreamlayers** [comment]
Jpegoptim has been changed now so it won't output progressive JPEG images by default with mozjpeg 3.1. In any case, my workaround for this issue is running jpegoptim twice on photos, first with `--all-progressive` and then with `--all-normal`.

**Comment by tjko** [comment]
Default behavior is to preserve the mode (progressive vs. non-progressive) in each JPEG image. That's why there is options --all-progressive and --all-normal so user can override this behavior if needed.
